src/components/Progress/Progress.tsx

```diff
@@ -0,0 +1,233 @@
+import React, {Component} from 'react';
+import {block} from '../utils/cn';
+import _ from 'lodash';
+
+import './Progress.scss';
+
+const b = block('progress');
+
+export type Theme = 'default' | 'success' | 'warning' | 'danger' | 'info' | 'misc';
+export type View = 'normal' | 'thin' | 'thinnest';
+export type Value = number;
+
+interface Stack {
+    value: Value;
+    color?: string;
+    title?: string;
+    theme?: Theme;
+    className?: string;
+    content?: React.ReactNode;
+}
+
+export interface ColorStops {
+    theme: Theme;
+    stop: number;
+}
+
+interface ProgressGeneralProps {
+    /** ClassName of element */
+    className?: string;
+}
+
+interface ProgressDefaultProps {
+    /** Text inside progress bar */
+    text: string;
+    /** Theme */
+    theme: Theme;
+    /** View. Text of progress bar is displayed in `normal` view only. */
+    view: View;
+}
+
+interface ProgressWithValue extends ProgressGeneralProps, Partial<ProgressDefaultProps> {
+    /** Current progress value. Available range is from 0 to 100. If `stack` property is passed `value` is not required and behaves as maxValue. */
+    value: Value;
+    /** Theme breakpoints. [Details](#colorstops) */
+    colorStops?: ColorStops[];
+    /** Alternative value of `colorStops`. Available range is from 0 to 100. */
+    colorStopsValue?: Value;
+}
+
+interface ProgressWithStack extends ProgressGeneralProps, Partial<ProgressDefaultProps> {
+    /** Configuration of composite progress bar. Not required if a `value` property is passed. [Details](#stack) */
+    stack: Stack[];
+    value?: Value;
+    /** ClassName of stack element */
+    stackClassName?: string;
+}
+
+export type ProgressProps = ProgressWithStack | ProgressWithValue;
+
+export class Progress extends Component<ProgressProps> {
+    static defaultProps: ProgressDefaultProps = {
+        text: '',
+        theme: 'default',
+        view: 'normal',
+    };
+
+    static isFiniteNumber(value: number): boolean {
+        return isFinite(value) && !isNaN(value);
+    }
+
+    static isBetween(value: number, min: number, max: number): boolean {
+        return value >= min && value <= max;
+    }
+
+    static getOffset(value: number): number {
+        return value < 100 ? value - 100 : 0;
+    }
+
+    static getValueFromStack(stack: Stack[]): number {
+        return _.sumBy(stack, (item: Stack) => item.value);
+    }
+
+    static isProgressWithStack(props: ProgressProps): props is ProgressWithStack {
+        return (props as ProgressWithStack).stack !== undefined;
+    }
+
+    render() {
+        const {view, className} = this.props;
+
+        return (
+            <div className={b({view}, className)}>
+                {this.renderText()}
+                {this.renderContent()}
+            </div>
+        );
+    }
+
+    private get theme(): Theme {
```

Comment:
Let's refactor to normal function `getTheme()`, getters are confusing

Reply:
What is the problem with getters?

Reply:
Accessing `this.theme` property calls the function, which can cause unwanted side effects (not in this case but overall)